src/components/map/Popup.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useMemo } from 'react';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+import './Popup.css';
+
+const Popup = (props, context) => {
+    const { className = '', coordinates, onClose, children } = props;
+    const { map } = context;
+    const container = useMemo(() => document.createElement('div'), []);
+
+    useEffect(() => {
+        container.className = className;
+        map.openPopup(container, coordinates, onClose);
+        return () => map.closePopup();
+    });
```

Comment:
I think this needs an empty dependency array (`useEffect(..., [])`), otherwise this will be called on every render (rather than just on mount)

Reply:
Solved by adding all dependencies to the array.